docker/Dockerfile

```diff
@@ -0,0 +1,25 @@
+ARG PYTHON_VERSION=3.9
+
+FROM docker.io/python:${PYTHON_VERSION}-slim
+
+# Install libmagic. We don't need to install libolm, because we're installing it with a
+# wheel from gitlab.matrix.org later, which comes with libolm already compiled.
+RUN apt-get update -qq && apt-get install -qq libmagic1
+
+# Copy the necessary project files into the image.
+COPY setup.cfg pyproject.toml /
+COPY matrix_content_scanner /matrix_content_scanner/
```

Comment:
You might want to add a .dockerignore file, so that only these files are sent as the docker build context (and not docs/test/scripts-dev etc). The Synapse repo should have an example.

It could also be overkill for a small project like this.

Reply:
It sounds like something that's nice to have and does make the amount of data sent to the build context from 215MB to 97kB when building locally (I assume it includes stuff like mypy cache etc since I'm running it on my machine), so let's add one.